tests/integration/test_posts.py

```diff
@@ -51,3 +52,12 @@ def test_post_detail(self, client: FlaskClient, post: Post) -> None:
         assert response.json['body'] == 'ccc'
         assert response.json['createdAt'] == datetime_to_rfc822_string(post.created_at)
         assert response.json['updatedAt'] == datetime_to_rfc822_string(post.updated_at)
+
+    def test_post_detail_not_found(self, client: FlaskClient, post: Post) -> None:
```

Comment:
I think since this isn't the main code path, it's better to leave testing this to the web layer unit tests